### DISCUSSION TITLE
Please test Adaptec 29320LPE

### DISCUSSION THREAD

**Discussion by jceastham** [geerlingguy/raspberry-pi-pcie-devices#556] [General]
Hi, as per subject, it would be good to know if Adaptec's ASC-29320LPE single lane PCIe SCSI HBA card works with the RPi-5.

**Comment by jnelson007** [comment]
Driver (aic7xxx.ko) doesn't exist in Raspberry Pi OS, but it does in Ubuntu 22.04. Driver crashes with "UBSAN: shift-out-of-bounds". Doesn't work and that's too bad, I had a project that would of sounded great with the clickity clackity of an old SCSI hard drive.
(HBA was tested and fully working under x86_64 Linux, issue is exclusive to Raspberry Pi)

UPDATE: I'm an idiot; Works out of the box on Ubuntu 22.04, must have dtparam=pciex1, dtparam=pciex1_gen=3, dtoverlay=pcie-32bit-dma Works as intended.

**Reply by 6by9** [reply]
If you have a Pi5 and the card, then rebuilding the kernel only takes about 15 minutes if using a USB drive rather than SD card.
https://www.raspberrypi.com/documentation/computers/linux_kernel.html#natively-build-a-kernel for the build steps.

https://www.raspberrypi.com/documentation/computers/linux_kernel.html#prepare-to-configure for changing the selected modules. 

Easiest to press "/" in menuconfig to search for an option. Enter "SCSI_AIC7XXX" and press return. It should list one result. Press "1" to jump straight to that menu option, and then space to toggle it to "M". 
Right arrow to select "Exit", and return. Return for "Exit". Right arrow to "Exit" and return. Return for "yes" at "Do you wish to save your new configuration?".
Run the build again, but do NOT do the `make bcm2712_defconfig` step (that would restore the default kernel configuration).

**Reply by jceastham** [reply]
> Driver (aic7xxx.ko) doesn't exist in Raspberry Pi OS, but it does in Ubuntu 22.04. 

> UPDATE: I'm an idiot; Works out of the box on Ubuntu 22.04, must have dtparam=pciex1, dtparam=pciex1_gen=3, dtoverlay=pcie-32bit-dma Works as intended.

Sorry, I'm confused.......have you had the 29320LPE running on a RPi-5 (via a PCIe x1 slot.....e.g. https://thepihut.com/products/pcie-slot-for-raspberry-pi-5-p02) ?

I'm happy enough compiling kernels......been doing that since kernel version 2.0.x :-)
